dashboard/app/models/script_level.rb

```diff
@@ -297,7 +297,10 @@ def summarize(include_prev_next=true)
       display_as_unplugged: display_as_unplugged
     }
 
-    summary[:progression] = progression if progression
+    if progression
+      localized_progression_name = I18n.t("data.progressions.#{progression}", default: progression)
+      summary[:progression] = localized_progression_name
```

Comment:
please store the translated value separately as `summary[:progression_display_name]` for two reasons:

1. the summary also gets used to populate the data we present to the curriculum writer in levelbuilder. this data should not be translated. see https://github.com/code-dot-org/code-dot-org/blob/d0a00bd5b11fc2a921dd1fdb67b28df0e3e0c287/dashboard/app/models/lesson.rb#L180-L181

2. the browser groups these together based on the value of the progression: https://github.com/code-dot-org/code-dot-org/blob/2e204359755b078196faa09c94c4e99c0f4fbf70/apps/src/code-studio/progressRedux.js#L707 If a translation error occurred such that two progressions shared the same non-empty value, we wouldn't want that to break our rendering logic.

following this suggestion, in order to get this to show up in the code studio UI, i think you would just need to:

* add display_name to this object: https://github.com/code-dot-org/code-dot-org/blob/2e204359755b078196faa09c94c4e99c0f4fbf70/apps/src/code-studio/progressRedux.js#L711

* use `progression.display_name` here: https://github.com/code-dot-org/code-dot-org/blob/ce87cd73a12fca5bf094c808be3d0598969c78ca/apps/src/templates/progress/ProgressLessonContent.jsx#L43

and then update any tests which break.

if you want to split it up, another option would be to end this PR with storing the new value on `summary[:progression_display_name]` , and come back to the UI part later.